lib/puppet/provider/package/gem.rb

```diff
@@ -48,7 +48,7 @@ def self.provider_command
   # Doing do defers the evaluation of provider suitability until all commands are evaluated.
 
   has_command(:gemcmd, 'gem') do
-    is_optional
+    is_optional?
```

Comment:
This looks to be a case where the `CommandDefiner` method is supposed to be private, but here we have a  provider calling the method, so it's effectively public API.

It's true the `gem` provider is builtin, but there isn't anything preventing a 3rd party module from doing the same (and often times type and provider code is cargo culted because of the lack of API docs). I'd recommend disabling the cop for the `is_optional` method in `provider.rb`